CHANGES.md

```diff
@@ -1,10 +1,142 @@
 
+### Fixed bugs that could lead to incorrect results
+
+- [#4178](https://github.com/gap-system/gap/pull/4178) Fixed bugs in `RestrictedPerm` with second argument a range
+
+### Fixed bugs that could lead to crashes
+
+- [#3965](https://github.com/gap-system/gap/pull/3965) Fix potential garbage collector crashes on 64bit ARM systems
+- [#4076](https://github.com/gap-system/gap/pull/4076) Fix an infinite loop in `BoundedRefinementEANormalSeries` if large factors could not be refined, fix protected option of `IsomorphismSimplifiedFpGroup`, improve documentation of `IsAutomorphismGroup`
+
+### Fixed bugs that could lead to error messages
+
+- [#3980](https://github.com/gap-system/gap/pull/3980) Fixed `Gcd` for rational polynomials
+
+### Other fixed bugs
+
+- [#3963](https://github.com/gap-system/gap/pull/3963) Provide automatic compression/decompression of filenames ending `.gz` (as is claimed for example in the documentation of `InputTextFile`)
+- [#3944](https://github.com/gap-system/gap/pull/3944) The error checking in `PartialPerm` has been corrected such that invalid inputs (numbers < 1) are detected
+- [#4006](https://github.com/gap-system/gap/pull/4006) Fix `gac` to ensure that binaries it creates on Linux can load and run, even if a GAP package with a compiled kernel extension (such as `IO`) is present
+
+### Improved and extended functionality
+
+- [#3790](https://github.com/gap-system/gap/pull/3790) Add further information to the library of simple groups
+- [#3840](https://github.com/gap-system/gap/pull/3840) Speed up garbage collection
+
+### Packages
+
+- [#4016](https://github.com/gap-system/gap/pull/4016) Improved `etc/Makefile.gappkg` (used by GAP packages that want to build a simple GAP kernel extension)
+
+### Fixes/improvements in the experimental way to allow 3rd party code to link GAP as a library (libgap)
+
+- [#4081](https://github.com/gap-system/gap/pull/4081) Enhance `GAP_ValueGlobalVariable` to supported automatic variables (see `DeclareAutoreadableVariables`)
+
+### Fixes and improvements for the **Julia** integration
+
+- [#4042](https://github.com/gap-system/gap/pull/4042) Avoid access to JuliaTLS members by useing `jl_threadid()` and `jl_get_current_task()` helpers, fix compiler constness warnings in weakptr.c
+- [#4053](https://github.com/gap-system/gap/pull/4053) Fix the logic for scanning tasks in the Julia GC
+- [#4058](https://github.com/gap-system/gap/pull/4058) Refine the logic for scanning Julia stacks
+- [#4071](https://github.com/gap-system/gap/pull/4071) Make the Julia GC threadsafe when used from GAP.jl
+
+### Other changes
+
+- [#3922](https://github.com/gap-system/gap/pull/3922) Build system: New feature to execute `BuildPackages.sh` in parallel mode by adding `--parallel`
+- [#4041](https://github.com/gap-system/gap/pull/4041) Build system: Fix `make check` in out-of-tree builds
+
+### New packages redistributed with GAP
+
+
+### Updated packages redistributed with GAP
+
```

Comment:
@alex-konovalov can you provide the missing data here?

Reply:
We should perhaps also highlight the data fix in the smallgrp package, see release notes at https://github.com/gap-packages/smallgrp/blob/master/CHANGES.md

Reply:
What about generally adding a link to the `CHANGES.md` files (if applicable) of updated and new packages?

Reply:
Maybe also mention the new release of CTblLib (@ThomasBreuer what do you think?) and TransGrp which now provides representatives for all transitive permutation groups of degree at most 47 (@hulpke is that correct?)

Reply:
```gap
for n in RecNames(old) do
  if not IsBound(new.(n)) then Print(old.(n).PackageName, " REMOVED\n"); continue; fi;
  if old.(n).Version <> new.(n).Version then
    Print(new.(n).PackageName, ": ", old.(n).Version, " -> ", new.(n).Version, "\n");
  fi;
od;
```
This currently gives me this (I can change the format which is being output easily)
```
GAPDoc: 1.6.3 -> 1.6.4
4ti2Interface: 2019.09.02 -> 2020.10-02
AutoDoc: 2019.09.04 -> 2020.08.11
Browse: 1.8.8 -> 1.8.11
CAP: 2019.06.07 -> 2020.10-01
CddInterface: 2020.01.01 -> 2020.06.24
ExamplesForHomalg: 2019.09.02 -> 2020.10-02
Gauss: 2019.09.02 -> 2020.10-02
GaussForHomalg: 2019.09.02 -> 2020.10-02
GeneralizedMorphismsForCAP: 2019.01.16 -> 2020.10-01
GradedModules: 2020.01.02 -> 2020.10-02
GradedRingForHomalg: 2020.01.02 -> 2020.10-02
HomalgToCAS: 2019.12.08 -> 2020.10-02
IO_ForHomalg: 2019.09.02 -> 2020.10-02
LinearAlgebraForCAP: 2019.01.16 -> 2020.10-01
LocalizeRingForHomalg: 2019.09.02 -> 2020.10-02
MatricesForHomalg: 2020.01.02 -> 2020.10-04
ModulePresentationsForCAP: 2019.01.16 -> 2020.10-01
Modules: 2019.09.02 -> 2020.10-02
MonoidalCategories: 2019.06.07 -> 2020.10-01
NConvex: 2019.12.10 -> 2020.11-04
NumericalSgps: 1.2.1 -> 1.2.2
PackageManager: 1.0 -> 1.1
PolymakeInterface REMOVED
RingsForHomalg: 2019.12.08 -> 2020.11-01
SCO: 2019.09.02 -> 2020.10-02
SmallGrp: 1.4.1 -> 1.4.2
ToolsForHomalg: 2019.09.02 -> 2020.10-03
ToricVarieties: 2019.12.05 -> 2021.01.12
XMod: 2.77 -> 2.82
XModAlg: 1.17 -> 1.18
CTblLib: 1.2.2 -> 1.3.1
curlInterface: 2.1.1 -> 2.2.1
Digraphs: 1.1.1 -> 1.3.1
ferret: 1.0.2 -> 1.0.3
HAP: 1.25 -> 1.29
homalg: 2019.09.01 -> 2020.10-02
IRREDSOL: 1.4 -> 1.4.1
json: 2.0.1 -> 2.0.2
kan: 1.29 -> 1.32
matgrp: 0.63 -> 0.64
Polycyclic: 2.15.1 -> 2.16
PrimGrp: 3.4.0 -> 3.4.1
profiling: 2.2.1 -> 2.3
QPA: 1.30 -> 1.31
Semigroups: 3.2.3 -> 3.4.0
singular: 2019.10.01 -> 2020.12.18
TransGrp: 2.0.5 -> 3.0
Wedderga: 4.9.5 -> 4.10.0
AGT: 0.1 -> 0.2
```

Reply:
Nice, saved my time, was about doing the same :) We may only count in the CHANGES the number of updated packages, like before, or put there this full list, I don't have a preference... Although adding it here will already have it ready for the release announcement. 

So, seems we have all we need for the packages section.

Reply:
> TransGrp which now provides representatives for all transitive permutation groups of degree at most 47 (@hulpke is that correct?)
Yes, but if we mention this, we should credit this list of groups to Derek Holt.

Reply:
@fingolfin could you please order packages alphabetically in the output?

Reply:
@ThomasBreuer what I meant above asking whether to mention CTblLIb: there were many packages updated, but I thought of mentioning only essential updates in data libraries. Here we have SmallGrp and TransGrp worth mentioning already, and there is an update CTblLib: 1.2.2 -> 1.3.1 ready for the new release, so is there anything in it worth highlighting?

Reply:
@alex-konovalov see the sorted list proposed in a comment below, and in the latest version of `CHANGES.md` in this branch